gradle-jdks/src/main/java/com/palantir/gradle/jdks/GradleWrapperPatcher.java

```diff
@@ -45,6 +49,9 @@ public abstract class GradleWrapperPatcher {
     private static final String GRADLEW_PATCH = "gradlew-patch.sh";
     private static final String COMMENT_BLOCK = "###";
     private static final String SHEBANG = "#!";
+    private static final ExceptionWithSuggestion REGENERATE_WRAPPER_PATCHER = new ExceptionWithSuggestion(
```

Comment:
This exception will have a fixed stacktrace based on when this class was created (will be at `GradleWrapperPatcher.<init>`) for this no matter which codepath actual happens when this stacktrace is thrown. Exceptions need to be made at the last possible moment to get accurate info.